### PR TITLE
DEV-7030 Update OpenAPI generator to 4.3.0

### PR DESCRIPTION
**Pull request by rizwansaeed**
Migrate to csharp-netcore
Remove C# templates
Update Python model

### CODE REVIEW

File: csharp/generate/generate.sh [hunk 1]
```diff
@@ -55,14 +55,15 @@ echo "generating sdk"
...
+    -g csharp-netcore \
     -o $sdk_output_folder \
     -c $config_file \
     -t $gen_root/templates \
 	--type-mappings dateorcutlabel=DateTimeOrCutLabel \
   --type-mappings double=decimal
 
-rm -rf $sdk_output_folder/.openapi-generator
+rm -rf $sdk_output_folder/.openapi-generator/
+rm -rf $sdk_output_folder/src/Lusid.Sdk.Test/
```
Comment (MikeMcGarry) [inline csharp/generate/generate.sh]: Why have you added in this line to remove the Tests?

File: python/generate/.openapi-generator-ignore [hunk 2]
```diff
@@ -6,4 +6,6 @@ lusid_README.md
 test-requirements.txt
 tox.ini
 README.md
-requirements.txt
\ No newline at end of file
+requirements.txt
+setup.cfg
+.gitlab-ci.yml
```
Comment (MikeMcGarry) [inline python/generate/.openapi-generator-ignore]: What are these new entries?

File: python/generate/templates/model.mustache [hunk 3]
```diff
@@ -62,7 +66,7 @@ class {{classname}}(object):
     }
 {{/discriminator}}
 
-    def __init__(self{{#vars}}, {{name}}{{#defaultValue}}={{{defaultValue}}}{{/defaultValue}}{{^defaultValue}}=None{{/defaultValue}}{{/vars}}):  # noqa: E501
+    def __init__(self{{#vars}}, {{name}}={{#defaultValue}}{{{defaultValue}}}{{/defaultValue}}{{^defaultValue}}None{{/defaultValue}}{{/vars}}, local_vars_configuration=None):  # noqa: E501
```
Comment (MikeMcGarry) [inline python/generate/templates/model.mustache]: I feel that the equals sign was moved and duplicated for a reason here but I can't remember why. Do you know if this doesn't cause any awkward issues having it moved to the front?

File: csharp/generate/generate.sh (hunk 1 above)
Comment (rizwansaeed) [reply csharp/generate/generate.sh]: its the autogenerated stub tests that get created, there isn't an option to not generate them (like some of the other generators) so explicitly deleting

File: python/generate/.openapi-generator-ignore (hunk 2 above)
Comment (rizwansaeed) [reply python/generate/.openapi-generator-ignore]: new files that get created that we don't use/need

File: python/generate/templates/model.mustache (hunk 3 above)
Comment (rizwansaeed) [reply python/generate/templates/model.mustache]: do you mean this was a change we made or is this from updating the template? tests all pass so I would presume its ok

**Pull request merged**